pyartifactory/objects.py

```diff
@@ -545,3 +571,148 @@ def delete(self, permission_name: str) -> None:
+                        "documents": (local_filename, f, "application/octet-stream"),
+                        "composite": "NONE",
+                    }
+                )
+                headers = {"Prefer": "respond-async", "Content-Type": form.content_type}
+                self._put(f"{artifact_path}", headers=headers, data=form)
+                logging.info(f"Artifact {local_filename} successfully deployed")
+                return self.properties(artifact_path)
+        except ArtifactDeployException:
+            logging.error(f"Cannot deploy artifact {local_filename}")
+            raise
+
+    def download(self, artifact_path: str) -> str:
+        """
+        :param artifact_path: Path to file in Artifactory
+        :return: File name
+        """
+        artifact_path = artifact_path.lstrip("/")
+        local_filename = artifact_path.split("/")[-1]
+        try:
+            with self._get(f"{artifact_path}", stream=True) as r:
+                with open(local_filename, "wb") as f:
+                    for chunk in r.iter_content(chunk_size=8192):
+                        if chunk:  # filter out keep-alive new chunks
+                            f.write(chunk)
+                            # f.flush()
+            logging.info(f"Artifact {local_filename} successfully downloaded")
+            return local_filename
+        except ArtifactDownloadException:
+            logging.error(f"Cannot download artifact {local_filename}")
+            raise
+
+    def properties(self, artifact_path: str) -> ArtifactPropertiesResponse:
+        """
+        :param artifact_path: Path to file in Artifactory
+        :return: Artifact properties
+        """
+        artifact_path = artifact_path.lstrip("/")
+        try:
+            r = self._get(f"api/storage/{artifact_path}?properties[=x[,y]]")
+            logging.info("Artifact Properties successfully retrieved")
+            return ArtifactPropertiesResponse(**r.json())
+        except ArtifactPropertiesException:
+            logging.error(f"Cannot retrieve artifact properties")
+            raise
+
+    def stats(self, artifact_path: str) -> ArtifactStatsResponse:
+        """
+        :param artifact_path: Path to file in Artifactory
+        :return: Artifact Stats
+        """
+        artifact_path = artifact_path.lstrip("/")
+        try:
+            r = self._get(f"api/storage/{artifact_path}?stats")
+            logging.info("Artifact stats successfully retrieved")
+            return ArtifactStatsResponse(**r.json())
+        except ArtifactPropertiesException:
+            logging.error(f"Cannot retrieve artifact stats")
+            raise
+
+    def copy(
+        self, artifact_current_path: str, artifact_new_path: str, dryrun: bool = False
+    ) -> bool:
+        """
+        :param artifact_current_path: Current path to file
+        :param artifact_new_path: New path to file
+        :param dryrun: Dry run
+        :return: True if the move is successful

```

Comment:
Maybe returning the properties of the new artifact would be more interesting than just a boolean?

Reply:
I'm not sure, when you do a copy, you only want to know wether it is successful or not.